src/node/blockstorage.cpp

```diff
@@ -684,10 +684,6 @@ bool BlockManager::UndoWriteToDisk(const CBlockUndo& blockundo, FlatFilePos& pos
 
     // Write undo data
     long fileOutPos = fileout.tell();
-    if (fileOutPos < 0) {
-        LogError("%s: ftell failed\n", __func__);
-        return false;
-    }
     pos.nPos = (unsigned int)fileOutPos;
```

Comment:
Does it still make sense for `AutoFile::tell()` to return a `int64_t` instead of a `uint64_t` or `unsigned int`?

Now that we've removed the invalid usages, maybe it's a good opportunity to transition `AutoFile` (and maybe `FlatFilePos`) to `std::optional<uint64_t> m_position;`

Reply:
> The advantage of uint64_t (and presumably the reason you're suggesting this) is because it signals that the number cannot be negative.

Yes, to make it stricter and closer aligned to the problem domain.

> signed types have the advantage that they actually model integers

But here we're modelling a size where potential negative values are confusing (especially since we're storing it as unsigned int going forward).
The negative values were already handled, it seems to me that they lost their usefulness after this change (checking for <0 after that change would warn).

> signed overflow is UB

We're already potentially down-casting the size to <64 bits - is overflow something we care about here?

Reply:
Would be nice to add an assertion / check here to make that fileOutPos is within range of unsigned int before casting it, to avoid silent truncation. But yes, out of scope of this PR as the conversion isn't introduced here.